src/app/component/product-input-form/product-input-form.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Product } from 'src/app/model/product.model';
+
+@Component({
+  selector: 'app-product-input-form',
+  templateUrl: './product-input-form.component.html',
+  styleUrls: ['./product-input-form.component.scss'],
+})
+export class ProductInputFormComponent implements OnInit {
+  @Input('product')
+  product?: Product | null
+
+  @Output('onSubmit')
+  submitEmitter = new EventEmitter<Product>();
+
+  @Output('onCancel')
+  cancelEmitter = new EventEmitter<void>();
+
+  productForm = this.fb.group({
```

Comment:
I'm guessing you are doing the separation for the Smart/Dumb Component, the form in this case should be in the smart component (you can pass it to the dumb component as an input)